react/week4/my-app/src/components/FetchUser.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useContext } from "react";
+import { stateContext } from "../App.js";
+
+function FetchUser() {
+  const {
+    query,
+    user,
+    setUser,
+    error,
+    setError,
+    loading,
+    setLoading
+  } = useContext(stateContext);
+
+  useEffect(() => {
+    const URL = `https://api.github.com/search/users?q=${query}`;
+    (async () => {
+      try {
+        const response = await fetch(`${URL}`);
+        const json = await response.json();
+        const result = json;
+        setUser(result);
+        setLoading(false);
```

Comment:
`setLoading` will be called only if neither `fetch` nor `json` methods fail. If one of them does, you will still have `loading` true, even though it will be finished.